### PR TITLE
refactor: Add LoadingView component to Select dropdown

### PR DESCRIPTION
**Pull request by MoizAdnan**
https://tsu.atlassian.net/browse/IR-2871

### CODE REVIEW

**Review by SYBIOTE — APPROVED** [review]

File: packages/ui/src/primitives/tailwind/Select/index.tsx [hunk 1]
```diff
@@ -155,17 +158,21 @@ const Select = <T extends OptionValueType>({
...
-              disabled ? 'opacity-50' : ''
-            }`}
-            onClick={() => {
-              if (!disabled) {
-                toggleDropdown()
-              }
-            }}
-          />
+          loading ? (
+            <LoadingView spinnerOnly className="mr-2 h-6 w-6" />
```
Comment (aditya-mitra) [inline packages/ui/src/primitives/tailwind/Select/index.tsx]: `LoadingView` here with a fixed set of props might affect the component's dynamic nature and is tightly coupled. For example, if we need to provide text to the `LoadingView` we will need to add another prop in `SelectProps` which increases prop complexity.

**Review by aditya-mitra — CHANGES_REQUESTED** [review]
Adding the `LoadingView` to `Select` makes the loading part tightly coupled.
We can instead pass an `endComponent` prop from `Select` down to `Input` component

File: packages/ui/src/primitives/tailwind/Select/index.tsx [hunk 2]
```diff
@@ -155,17 +157,21 @@ const Select = <T extends OptionValueType>({
...
-            }`}
-            onClick={() => {
-              if (!disabled) {
-                toggleDropdown()
-              }
-            }}
-          />
+          endComponent ? (
+            endComponent
+          ) : (
```
Comment (aditya-mitra) [inline packages/ui/src/primitives/tailwind/Select/index.tsx]: one liner 💡

```suggestion
          endComponent ?? (
```

**Review by aditya-mitra — APPROVED** [review]
Looks good ✅

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**